w06h03/test/pgdp/ds/behavior/GraphTest.java

```diff
@@ -31,7 +33,9 @@ final void nodeShouldBeNeighborIfConnectionWithEdge() {
         };
         graph.addEdge(1, expectedNeighbors[0]);
         graph.addEdge(1, expectedNeighbors[1]);
-        assertArrayEquals(expectedNeighbors, graph.getAdj(1));
```

Comment:
For future tests, you may consider asserting that two Collections (e.g. Set.of(adj) or HashSets) are equal. This may improve the performance and may be more intuitive.

Reply:
Yeah of course 👍

Reply:
> Yeah of course 👍

Nevermind... That would be a lot worse (arrays of primitive types can't be converted into a set by calling `Set.of()`...):

```java
Assertions.assertEquals(Set.of(expectedNeighbors), Arrays.stream(graph.getAdj(1)).boxed().collect(Collectors.toSet()));
```

Reply:
correct, you would have to properly box them

Reply:
However, for large arrays, you will need linear time to create a set while sorting an array will take time O(n log n) which may be slower. Furthermore, if there is no requirement of sortedness it might be more intuitive to argue about sets :)